src/store/Workspaces/cheWorkspaces/index.ts

```diff
@@ -266,6 +275,39 @@ export const actionCreators: ActionCreators = {
     }
   },
 
+  restartWorkspace: (workspace: che.Workspace): AppThunk<KnownAction, Promise<void>> => async (dispatch, getState): Promise<void> => {
+    const defer: IDeferred<void> = getDefer();
+    const toDispose = new DisposableCollection();
+    const onStatusChangeCallback = async (status: string) => {
+      if (status !== WorkspaceStatus.STOPPED && status !== WorkspaceStatus.ERROR) {
+        return;
+      }
+      try {
+        await dispatch(actionCreators.startWorkspace(workspace));
+        defer.resolve();
+      } catch (e) {
+        defer.reject(e);
+      }
+      toDispose.dispose();
+    };
+    if (workspace.status === WorkspaceStatus.STOPPED || workspace.status === WorkspaceStatus.ERROR) {
+      await onStatusChangeCallback(workspace.status);
+    } else {
+      subscribeToStatusChange(workspace, dispatch);
```

Comment:
All workspaces at this moment are already subscribed to status changes.
```suggestion
```

Reply:
I have removed it.